design/authenticators_api.md

```diff
@@ -0,0 +1,187 @@
+but the action they need it for is different. 
+
+We can learn 2 things from the above:
+
+1. The Conjur user needs to authenticate according to the response. So this request must be applicable **without** a Conjur access token. 
+1. The Conjur operator needs to know which actions are needed for completing the configuration,
+which may consist sensitive data. So this request must be applicable **only with** a Conjur access token.
+
+This leads us to 2 API endpoints:
+
+1. `/authenticators` (for Conjur users)
+2. `/authenticators/health` (for Conjur operators)
+
+### Process Logic
+
+In the following example, the authenticators `authn` & `authn-1` are properly configured
+and `authn-2` is not whitelisted in the ENV.
+
+#### `/authenticators` Endpoint
+
+- A Conjur user runs an `/authenticators` request:\
+`GET /authenticators`
+- The user gets a response with code 200 with the following body:
+```
+{
+   “authenticators”:
+   [
+      "authn",
+      "authn-1"
+   ]
+}
+```
+
+#### `/authenticators/health` Endpoint
+
+- A Conjur operator logs into Conjur (in any authn method) and receives an access token
+- The operator runs an `/authenticators/health` request with the given access token:\
+`GET /authenticators/health`
+- The operator gets a response with code 500 (more info [here](authenticators_health_api.md#response-code-for-unhealthy-authenticators)) with the following body:
+```
+{
+   "authenticators":
+   {
+      "authn": {
+        "status": "ok"
+      },
+      "authn-1": {
+        "status": "ok"
+      },
+      "authn-2": {
+        "status": "error",
+        "error": "Authentication::Security::NotWhitelisted",
+        "code": "CONJ00004E",
+        "message": "'authn-2' is not whitelisted in CONJUR_AUTHENTICATORS",
+      }
+   },
+   "status": "error"
+}
+```
+- The operator understands the issue from the response and fixes it
+- The operator runs the request once again until he gets a healthy response
+
+#### `/info` endpoint
+
+Another endpoint we should address is `evoke`'s `/info` endpoint, which contains the data of the `/authenticators` endpoint
+(with other info such as release, version, etc.). This endpoint calls `http://localhost/authenticators` and adds the output to the info
+json. Having the same data in 2 different places is not optimal as it can confuse the user (and developers), so it would make sense to move the
+implementation of the `/authenticators` endpoint to `evoke`, and remove the `/authenticators` endpoint.
+However, as `evoke` is not part of the OSS, we will leave the implementation in the `/authenticators` endpoint and will not change the implementation in `evoke`.
+
+### Security
+
+#### `/authenticators` Endpoint
+
+Although the response reveals which authenticators are configured and this request is prone
+for brute-force attacks, the value of this endpoint for the user is high and can
+provide the information needed for authentication. Furthermore, we don't say _why_
```

Comment:
from my reading above this endpoint only says the authenticators that are validly configured for use. can you explain more why you note:

> we don't say _why_ the authenticator is invalid

given that it doesn't show invalid authenticators at all, I do not follow how that is an argument for making the endpoint open (though I do agree it is fine for this endpoint to be open)

Reply:
i was just stating that saying why an authenticator is invalid is insecure and because we don't do that then it is ok.